tests/test_stainaugment.py

```diff
@@ -0,0 +1,83 @@
+"""Tests for stain augmentation code."""
+
+import pathlib
+
+import albumentations as alb
+import numpy as np
+import pytest
+
+from tiatoolbox.data import stainnorm_target
+from tiatoolbox.tools.stainaugment import StainAugmentation
+from tiatoolbox.tools.stainnorm import get_normaliser
+from tiatoolbox.utils.misc import imread
+
+
+def test_stainaugment(source_image, norm_vahadane):
+    """Test functionality of the StainAugmentation class."""
+    source_img = imread(pathlib.Path(source_image))
+    target_img = stainnorm_target()
+    vahadane_img = imread(pathlib.Path(norm_vahadane))
+
+    # Test invalid method in the input
+    with pytest.raises(ValueError, match=r".*Unsupported stain extractor method.*"):
+        _ = StainAugmentation(method="mosi")
```

Comment:
I like how you put your nickname, but shall we put something more clear like ‘invalid’

Reply:
thanks for seeing this :-D I forget to change it.